### PR TITLE
71 user roles

### LINKED ISSUE #71: User roles
We should think about how we want to implement roles, so we can start implementing authorization.

Currently we need roles for:
- global scope (`admin`, `user`, ...?)
- member roles in organizations (`admin`, `write`, `read`, `none`)
- member roles in repositories (not implemented in the models yet)

I think for the first one, we should add a field to the `user` model with an enum of our supported global roles (even if we only have `admin` and `user`, I think we should still use an enum instead of a boolean field, so it can be easily extended).

The second one should be implemented in the join-table `organizations_members`, again using an enum for the available roles. I think we would then need to change the `many_to_many` relationship (or add another `one_to_many`) to be able to access the role data.

The third one would be similar to the second one.

### PR DESCRIPTION
**Pull request by ebolloff**
Should fix #71.
* [x] Change methods (`user.rb`) to associations because of performance

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -95,6 +95,68 @@
...
+    let!(:organization1) { create :organization }
+    let!(:organization2) { create :organization }
+    let!(:user2) { create :user }
+    let!(:repository1) { create(:repository, owner: subject) }
+    let!(:repository2) { create(:repository, owner: organization1) }
+    let!(:repository3) { create(:repository, owner: organization2) }
+    let!(:repository4) { create(:repository, owner: user2) }
+    let!(:repository5) { create(:repository, owner: organization2) }
+
+    before do
```
Comment (ebolloff) [inline spec/models/user_spec.rb]: Line is not too long, I will ignore this.

File: db/migrate/20160902111740_initial_schema.rb [hunk 2]
```diff
@@ -19,13 +19,16 @@
     end
 
     # User is an OrganizationalUnit
+    create_enum :global_role,
+    %w(admin user)
```
Comment (eugenk) [inline db/migrate/20160902111740_initial_schema.rb]: Please indent this two more spaces.

File: db/migrate/20160902111740_initial_schema.rb [hunk 3]
```diff
@@ -35,10 +38,13 @@
       column :description, String
     end
 
-    create_table :organizations_members do
+    create_enum :organization_role,
+    %w(admin write read)
```
Comment (eugenk) [inline db/migrate/20160902111740_initial_schema.rb]: Please indent this two more spaces.

File: db/migrate/20160902111740_initial_schema.rb [hunk 4]
```diff
@@ -59,6 +65,15 @@
       column :content_type, :repository_content_type
     end
 
+    create_enum :repository_role,
+    %w(admin write read)
```
Comment (eugenk) [inline db/migrate/20160902111740_initial_schema.rb]: Please indent this two more spaces.

**Pull request merged**